File: Circular_Linked_list/CircularSinglyLinkedListCreateNode.java
// File Name: CircularSinglyLinkedListCreateNode.java
// Program to create a node in a circular singly linked list

class Node {
    int data;
    Node next;

    Node(int data) {
        this.data = data;
        this.next = this;   // Circular link
    }
}

public class CircularSinglyLinkedListCreateNode {

    // Function to create a new node
    static Node createNode(int value) {
        return new Node(value);
    }

    public static void main(String[] args) {
        Node head = createNode(10);
    }
}

File: Circular_Linked_list/circular_singly_linked_list_create_node.c
// File Name: circular_singly_linked_list_create_node.c
// Program to create a node in a circular singly linked list

#include <stdio.h>
#include <stdlib.h>

// Definition of node
struct node {
    int data;
    struct node *next;
};

// Function to create a new node
struct node* create_node(int value) {
    struct node *newnode = (struct node *)malloc(sizeof(struct node));
    if (newnode == NULL) {
        printf("Memory allocation failed\n");
        return NULL;
    }
    newnode->data = value;
    newnode->next = newnode;   // Circular link
    return newnode;
}

int main() {
    struct node *head = create_node(10);
    return 0;
}
